functions/assert_metadata_os.pp

```diff
@@ -0,0 +1,57 @@
+# Fails a compile if the client system is not compatible with the module's
```

Comment:
Why do we need to explicitly fail if a particular module doesn't support an OS? We already codify that we only support certain OSs in our metadata.json and in simp-core/the iso. Why should we punish those who try different ones?